### ISSUE TITLE
inspect.iscorutinefunction() returns False for unittest.mock.AsyncMock instances

### ISSUE THREAD

**Issue by 8e7ee72d-6875-4619-98a4-d106d76a25e3** [python/cpython#84753]
BPO | [40573](https://bugs.python.org/issue40573)
--- | :---
Nosy | @moriyoshi, @1st1, @lisroach, @tirkarthi, @iritkatriel

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2020-05-09.03:14:11.981>
labels = ['3.8', 'type-bug', 'library', '3.9', '3.10']
title = 'inspect.iscorutinefunction() returns False for unittest.mock.AsyncMock instances'
updated_at = <Date 2020-08-25.16:22:34.440>
user = 'https://github.com/moriyoshi'
```

bugs.python.org fields:
```python
activity = <Date 2020-08-25.16:22:34.440>
actor = 'iritkatriel'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2020-05-09.03:14:11.981>
creator = 'moriyoshi'
dependencies = []
files = []
hgrepos = []
issue_num = 40573
keywords = []
message_count = 3.0
messages = ['368497', '368498', '375886']
nosy_count = 5.0
nosy_names = ['moriyoshi', 'yselivanov', 'lisroach', 'xtreak', 'iritkatriel']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue40573'
versions = ['Python 3.8', 'Python 3.9', 'Python 3.10']
```

</p></details>

**Comment by graingert** [comment]
looks like this issue is `mock.AsyncMock()` doesn't pass `inspect.isfunction`:

```python
from unittest import mock
import inspect

m = mock.AsyncMock()
with mock.patch("inspect.isfunction", new=lambda x: True):
    assert inspect.iscoroutinefunction(m)
```

**Comment by ambv** [comment]
Fixed with #94050 and backported to 3.10 and 3.11. Thanks! ✨ 🍰 ✨
